src/datanode/src/instance.rs

```diff
@@ -220,6 +221,24 @@ impl Instance {
         Ok(())
     }
 
+    pub async fn shutdown(&self) -> Result<()> {
+        if let Some(heartbeat_task) = &self.heartbeat_task {
+            heartbeat_task
+                .close()
+                .await
+                .map_err(BoxedError::new)
+                .context(ShutdownInstanceSnafu)?;
+        }
+
+        self.sql_handler
+            .close()
+            .await
+            .map_err(BoxedError::new)
+            .context(ShutdownInstanceSnafu)?;
+
+        Ok(())
```

Comment:
```suggestion
        self.sql_handler
            .close()
            .await
            .map_err(BoxedError::new)
            .context(ShutdownInstanceSnafu)
```

Reply:
> Suggested change
>         self.sql_handler
>             .close()
>             .await
>             .map_err(BoxedError::new)
>             .context(ShutdownInstanceSnafu)?;
>         Ok(())
>         self.sql_handler
>             .close()
>             .await
>             .map_err(BoxedError::new)
>             .context(ShutdownInstanceSnafu)

It seems like leading to a mismatched types error:
```
mismatched types
expected enum `std::result::Result<(), _>`
   found enum `std::result::Result<Vec<()>, _>`
```

Reply:
Goit

Reply:
> > Suggested change
> > self.sql_handler
> > .close()
> > .await
> > .map_err(BoxedError::new)
> > .context(ShutdownInstanceSnafu)?;
> > Ok(())
> > self.sql_handler
> > .close()
> > .await
> > .map_err(BoxedError::new)
> > .context(ShutdownInstanceSnafu)
> 
> It seems like leading to a mismatched types error:
> 
> ```
> mismatched types
> expected enum `std::result::Result<(), _>`
>    found enum `std::result::Result<Vec<()>, _>`
> ```

My bad, not this one